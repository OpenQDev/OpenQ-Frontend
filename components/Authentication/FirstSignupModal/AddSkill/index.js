import React, { useContext } from 'react';
import StoreContext from '../../../../store/Store/StoreContext';
import Cross from '../../../svg/cross';

const AddSkill = ({ category, rolesInCategoriesState }) => {
  const [rolesInCategories, setRolesInCategories] = rolesInCategoriesState;
  const theseRoles = rolesInCategories[category];
  const [appState, appDispatch] = useContext(StoreContext);
  const { accountData } = appState;

  const updateApiAndState = async (newRoles, category) => {
    const newRolesAndCategory = { ...rolesInCategories, [category]: newRoles };
    const userValues = accountData.github
      ? {
          github: accountData.github,
          ...newRolesAndCategory,
        }
      : {
          email: accountData.email,
          ...newRolesAndCategory,
        };
    const { updateUser } = await appState.openQPrismaClient.updateUser(userValues);
    appDispatch({ type: 'UPDATE_ACCOUNT_DATA', payload: updateUser });
    setRolesInCategories(newRolesAndCategory);
  };

  const removeRole = (e, removedRole) => {
    e.preventDefault();
    const newRoles = theseRoles.filter((role) => role !== removedRole);
    updateApiAndState(newRoles, category);
  };
  return (
    <>
      {theseRoles?.map((role, index) => {
        return (
          <button
            className={`flex items-center gap-2 py-0.5 btn-default w-fit m-1`}
            key={index}
            onClick={(e) => removeRole(e, role)}
          >
            {role}
            <Cross />
          </button>
        );
      })}
    </>
  );
};

export default AddSkill;
