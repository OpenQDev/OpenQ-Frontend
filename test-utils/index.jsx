// Third party
import React from 'react';
// Cusotm
import { render, renderHook } from '@testing-library/react';
import { cleanup } from '@testing-library/react';

import StoreProvider from '../store/Store/TestStoreProvider';
import AuthProvider from '../store/AuthStore/TestAuthProvider';
import FundProvider from '../components/FundBounty/FundStore/TestFundProvider';
import TokenProvider from '../components/TokenSelection/TokenStore/TokenProvider';

// Add in any providers here if necessary:
// test-utils.js
// (ReduxProvider, ThemeProvider, etc)
afterEach(() => {
  cleanup();
});

const customRender = (ui, options = {}, storeProps, authProps) => {
  const IntersectionObserverMock = vi.fn(() => ({
    disconnect: vi.fn(),
    observe: vi.fn(),
    takeRecords: vi.fn(),
    unobserve: vi.fn(),
  }));
  vi.stubGlobal('IntersectionObserver', IntersectionObserverMock);
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <TokenProvider>{children}</TokenProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return render(ui, { wrapper: Providers, ...options });
};
const customHookRender = (hook, options = {}, storeProps, authProps, fundProps) => {
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <FundProvider StoreProps={fundProps}>
            <TokenProvider>{children}</TokenProvider>
          </FundProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return renderHook(hook, { wrapper: Providers, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, customHookRender as hookRender };
