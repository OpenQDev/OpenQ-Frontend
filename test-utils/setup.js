import { expect, beforeEach } from 'vitest';
import { cleanup } from '@testing-library/react';
import matchers from '@testing-library/jest-dom/matchers';

expect.extend(matchers);
beforeEach(() => {
  cleanup();
});
process.on('unhandledRejection', (reason) => {
  // eslint-disable-next-line no-console
  console.log(`FAILED TO HANDLE PROMISE REJECTION`);
  throw reason;
});
