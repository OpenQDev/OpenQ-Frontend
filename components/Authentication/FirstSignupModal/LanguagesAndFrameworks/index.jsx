import React, { useEffect, useState, useContext } from 'react';
import AddSkill from '../AddSkill';
import StoreContext from '../../../../store/Store/StoreContext';

const LanguagesAndFrameworks = ({ user }) => {
  const [appState, appDispatch] = useContext(StoreContext);
  const { accountData } = appState;
  const languages = [
    'python',
    'java',
    'c++',
    'c#',
    'javascript',
    'ruby',
    'swift',
    'go',
    'scala',
    'kotlin',
    'perl',
    'php',
    'pascal',
    'objective-c',
    'r',
    'rust',
    'dart',
    'elixir',
    'erlang',
    'f#',
    'haskell',
    'html',
    'css',
    'clojure',
    'crystal',
    'elm',
    'groovy',
    'julia',
    'nim',
    'ocaml',
    'red',
    'typescript',
    'solidity',
    'scheme',
    'smalltalk',
    'sql',
  ];
  const frameworks = [
    'django',
    'flask',
    'rails',
    'laravel',
    'express',
    'spring',
    'asp.net',
    'react',
    'vue',
    'angular',
    'ember',
    'meteor',
    'sails',
    'sinatra',
    'express.js',
    'koa',
    'hapi',
    'socket.io',
    'next.js',
    'nuxt.js',
    'strapi',
    'prisma',
    'nestjs',
    'adonis',
    'loopback',
    'svelte',
    'tailwind',
    'bulma',
    'element-ui',
    'ant-design',
    'bootstrap',
    'material-ui',
    'semantic-ui',
  ];
  const categories = {
    languages: languages,
    frameworks: frameworks,
    otherRoles: [],
  };

  const updateApiAndState = async (newRoles, accountData, appState) => {
    const newRolesAndCategory = { ...rolesInCategories, ...newRoles };
    const userValues = accountData.github
      ? {
          github: accountData.github,
          ...newRolesAndCategory,
        }
      : {
          email: accountData.email,
          ...newRolesAndCategory,
        };
    const { updateUser } = await appState.openQPrismaClient.updateUser(userValues);
    appDispatch({ type: 'UPDATE_ACCOUNT_DATA', payload: updateUser });
    setRolesInCategories(newRolesAndCategory);
  };

  const rolesInCategoriesState = useState({
    languages: [],
    frameworks: [],
    otherRoles: [],
  });
  const [rolesInCategories, setRolesInCategories] = rolesInCategoriesState;
  const [inputValue, setInputValue] = useState('');

  const handleKeypress = (e, inputs) => {
    //it triggers by pressing the enter key
    if (e.key === 'Enter' && e.target.value != '') {
      checkCategory(inputs);
    }
  };
  useEffect(() => {
    const addUsersLanguages = async () => {
      await checkCategory(user.recentLanguages?.map((lang) => lang.toLowerCase()));
    };
    addUsersLanguages();
  }, [user]);
  const handleInputChange = (e) => {
    const { value } = e.target;
    setInputValue(value);
    // setValidUsername(true); Check if lang or framework here
  };
  const checkCategory = async (inputs) => {
    const allSkils = Object.values(rolesInCategories).flat();
    const filteredInputs = inputs.filter((input) => !allSkils.includes(input));
    if (!filteredInputs.length) return;
    let updateObj = { ...rolesInCategories };
    const createUpdateObj = (nameOfCategory, updateObj, input) => {
      updateObj = { ...updateObj, [nameOfCategory]: [...updateObj[nameOfCategory], input] };
      return updateObj;
    };
    const categoriesArr = Object.entries(categories);
    for (const input of filteredInputs) {
      let matches = false;
      for (const category of categoriesArr) {
        const nameOfCategory = category[0];
        const listOfRolesInCategory = category[1];
        if (listOfRolesInCategory.includes(input.toLowerCase())) {
          updateObj = createUpdateObj(nameOfCategory, updateObj, input);
          matches = true;
        }
      }
      if (!matches) updateObj = createUpdateObj('otherRoles', updateObj, input);
    }
    await updateApiAndState(updateObj, accountData, appState);

    setInputValue('');
  };
  return (
    <>
      <p className='font-semibold pt-4 py-2'>Tell us about your skills</p>
      <div className='flex flex-wrap'>
        {Object.entries(rolesInCategories).map((category) => {
          const nameOfCategory = category[0];
          return (
            <AddSkill
              rolesInCategoriesState={rolesInCategoriesState}
              key={category}
              category={nameOfCategory}
              setInputValue={setInputValue}
            />
          );
        })}
      </div>
      <div className='flex justify-center gap-2 mt-2'>
        <input
          className='flex input-field w-full'
          placeholder='Programming languages, frameworks...'
          onChange={handleInputChange}
          onKeyDown={(e) => handleKeypress(e, [inputValue])}
          value={inputValue}
        />
        <button onClick={() => checkCategory([inputValue])} className='btn-primary'>
          Add
        </button>
      </div>
    </>
  );
};

export default LanguagesAndFrameworks;
