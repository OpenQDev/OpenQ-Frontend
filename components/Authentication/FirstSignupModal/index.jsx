import React, { useContext, useEffect, useState } from 'react';
import ModalLarge from '../../Utils/ModalLarge';
import Image from 'next/image';
import AuthContext from '../../../store/AuthStore/AuthContext';
import SelectItem from '../SelectItem';
import SignOut from '../SignOut';
import StoreContext from '../../../store/Store/StoreContext';
import ToolTipNew from '../../Utils/ToolTipNew';
import Username from '../../User/OverviewTab/Username/index';
import LanguagesAndFrameworks from './LanguagesAndFrameworks';
import { useRouter } from 'next/router';
import FreelancerDetails from '../../User/InvoicingDetailsTab/FreelancerDetails';

const FirstSignupModal = ({ closeModal, setShowModal, user }) => {
  const router = useRouter();
  const [authState] = useContext(AuthContext);
  const [appState] = useContext(StoreContext);
  const [subscribe, setSubscribe] = useState(false);
  const [email, setEmail] = useState(appState.accountData.email);
  const [validEmail, setValidEmail] = useState(appState.accountData.email);
  const [disabled, setDisabled] = useState(false);
  const roles = [
    'Engineer',
    'Founder',
    'Product Manager',
    'Project Manager',
    'QA',
    'Designer',
    'Student',
    'Marketing',
    'Other',
  ];
  const products = ['Marketplace', 'Hackathon', 'DRM', 'Not sure'];

  useEffect(() => {
    if (subscribe && !validEmail) {
      setDisabled(true);
    } else {
      setDisabled(false);
    }
  }, [subscribe, validEmail]);

  async function submit() {
    if (subscribe && validEmail) {
      try {
        const formData = new FormData();
        formData.append('api_key', process.env.NEXT_PUBLIC_CONVERTKIT_API_KEY);
        formData.append('email', validEmail);
        const response = await fetch('https://api.convertkit.com/v3/forms/3697685/subscribe', {
          method: 'POST',
          body: formData,
        });
        await response.json();
      } catch (err) {
        appState.logger.error(err, appState.accountData.id, 'FirstSignupModal.js1');
      }
    }
    router.reload(`/user/${appState.accountData.id}`);
  }

  function handleChange(e) {
    const mail = e.target.value;
    setEmail(mail);
    setValidEmail(appState.utils.emailRegex(mail) && mail);
  }

  const btn = (
    <ToolTipNew hideToolTip={!disabled} toolTipText={'Please enter a valid email address to subscribe.'}>
      <button
        className={`btn-primary ${disabled && 'cursor-not-allowed'}`}
        onClick={() => submit()}
        disabled={disabled}
      >
        Continue
      </button>
    </ToolTipNew>
  );

  return (
    <ModalLarge
      isWalletConnect={true}
      title={
        <div className='flex justify-between text-sm'>
          <div>
            <Image src='/openq-logo-with-text.png' width={150} height={150} alt='openq-logo' />
          </div>
          <div className='flex flex-col md:flex-row items-center md:gap-4 pr-8'>
            <div className=''>{authState.login}</div>
            <SignOut
              styles={
                ' border-none bg-transparent text-[#1f6feb] md:hover:underline hover:bg-transparent rounded-none justify-start'
              }
              hidePropic={true}
            />
          </div>
        </div>
      }
      footerRight={btn}
      setShowModal={setShowModal}
      resetState={closeModal}
    >
      <div className='mx-4 p-4'>
        <div>
          <p className='text-2xl font-semibold  py-2'>Hello {authState.login}</p>
          <p className='py-2'>If you're coming here as an EthDenver Buidlathon winner, congratulations!</p>
          <p className='py-2'>Let's get you paid.</p>
          <p className='py-2'>Enter your email to stay up to date with next steps in claiming your award.</p>
          <p className='py-2'>
            OpenQ and EthDenver will use this email to tell you when you uploaded forms have been approved, after which
            you can claim right away.
          </p>
        </div>
        <section className='flex flex-col gap-3 pb-6'>
          <FreelancerDetails slim={true} emailOnly={true} />
        </section>
        <p className='text-2xl font-semibold  py-2'>Now, tell us a bit about yourself</p>
        <div className='flex gap-2'>
          <p className='font-semibold py-2'>Username:</p>
          <Username user={appState.accountData} firstSignup={true} />
        </div>
        <p className='font-semibold py-2'>What is your role?</p>
        <SelectItem fieldName={'otherRoles'} items={roles} />
        <LanguagesAndFrameworks user={user} />
        <p className='font-semibold pt-4 py-2'>Which OpenQ products are you interested in?</p>
        <SelectItem fieldName={'interests'} items={products} />
        <div className='flex flex-col pt-8 py-2'>
          <div className='flex items-center justify-start gap-2'>
            <input type='checkbox' className='checkbox' onChange={() => setSubscribe(!subscribe)}></input>
            <p className='font-semibold'>Join hundreds of professionals learning with OpenQ</p>
          </div>

          <div className='flex flex-col pt-4 text-muted gap-2'>
            {subscribe && !appState.acountData?.email && (
              <input
                className='input-field w-full'
                placeholder='Enter your email'
                size='sm'
                value={email}
                onChange={(e) => handleChange(e)}
              />
            )}
            <p>
              Curated emails on the latest software and web3 topics, product updates, events and new opportunities for
              your career.
            </p>
          </div>
        </div>
      </div>
    </ModalLarge>
  );
};
export default FirstSignupModal;
