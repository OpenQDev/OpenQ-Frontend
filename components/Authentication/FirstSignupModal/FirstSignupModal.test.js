import { render } from '../../../test-utils';
import React from 'react';
import FirstSignupModal from '.';
import nextRouter from 'next/router';

describe('FirstSignupModal', () => {
  nextRouter.useRouter = jest.fn();
  nextRouter.useRouter.mockImplementation(() => ({
    query: { type: null },

    prefetch: jest.fn(() => {
      return { catch: jest.fn };
    }),
  }));
  const user = {
    id: '63c58870f2943bf006029305',
    renderError: '',
    avatarUrl: 'https://avatars.githubusercontent.com/u/75732239?u=285a48b76fb74f9487a3f595002bdbd69522a151&v=4',
    __typename: 'User',
    name: 'Anya',
    login: 'ArcAnya',
    websiteUrl: 'arcanya.com',
    url: 'https://github.com/ArcAnya',
    twitterUsername: null,
    recentLanguages: ['Dockerfile', 'JavaScript', 'CSS', 'TypeScript', 'Mustache'],
    onChainAddress: null,
    github: 'MDQ6VXNlcjc1NzMyMjM5',
    devRoles: [],
    otherRoles: ['ohia'],
    interests: [],
    languages: ['dockerfile', 'javascript', 'css', 'typescript', 'mustache', 'rust'],
    frameworks: [],
  };
  it('should match initial DOM Snapshot', () => {
    const { asFragment } = render(<FirstSignupModal user={user} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
