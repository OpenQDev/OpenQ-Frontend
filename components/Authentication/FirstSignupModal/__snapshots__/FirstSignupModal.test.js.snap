// vi Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstSignupModal should match initial DOM Snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="flex justify-center items-center overflow-x-hidden overflow-y-auto fixed inset-0 z-50 z-[51]"
    >
      <div
        class="flex w-full md:w-[640px] h-full md:h-[600px] border border-gray-700 md:rounded-sm bg-[#161B22]"
      >
        <div
          class="flex flex-col relative w-full"
        >
          <button
            class="absolute top-4 right-4 cursor-pointer"
            data-testid="cross"
          >
            <svg
              class="fill-muted hover:fill-blue-400 w-4 h-4"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <div
            class="py-2 px-4 border-b border-gray-700 text-2xl w-full"
          >
            <div
              class="flex justify-between text-sm"
            >
              <div>
                <img
                  alt="openq-logo"
                  data-nimg="1"
                  decoding="async"
                  height="150"
                  loading="lazy"
                  src="/_next/image?url=%2Fopenq-logo-with-text.png&w=384&q=75"
                  srcset="/_next/image?url=%2Fopenq-logo-with-text.png&w=256&q=75 1x, /_next/image?url=%2Fopenq-logo-with-text.png&w=384&q=75 2x"
                  style="color: transparent;"
                  width="150"
                />
              </div>
              <div
                class="flex flex-col md:flex-row items-center md:gap-4 pr-8"
              >
                <div
                  class=""
                />
                <button
                  class="flex   btn-default hover:border-[#8b949e] hover:bg-[#30363d] w-full  border-none bg-transparent text-[#1f6feb] md:hover:underline hover:bg-transparent rounded-none justify-start "
                >
                  <div
                    class="flex flex-row justify-center items-center space-x-3"
                  >
                    <div
                      class="py-1"
                    >
                      <svg
                        aria-hidden="true"
                        class="w-4 h-4 ml-2 "
                        fill="currentColor"
                        focusable="false"
                        height="16"
                        role="img"
                        style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
                        viewBox="0 0 16 16"
                        width="16"
                      >
                        <path
                          d="M2 2.75C2 1.784 2.784 1 3.75 1h2.5a.75.75 0 0 1 0 1.5h-2.5a.25.25 0 0 0-.25.25v10.5c0 .138.112.25.25.25h2.5a.75.75 0 0 1 0 1.5h-2.5A1.75 1.75 0 0 1 2 13.25Zm10.44 4.5-1.97-1.97a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l3.25 3.25a.75.75 0 0 1 0 1.06l-3.25 3.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.97-1.97H6.75a.75.75 0 0 1 0-1.5Z"
                        />
                      </svg>
                    </div>
                    <div>
                      Sign Out
                    </div>
                  </div>
                </button>
              </div>
            </div>
          </div>
          <div
            class="w-full h-full md:h-[500px] overflow-x-hidden overflow-y-auto"
          >
            <div
              class="mx-4 p-4"
            >
              <div>
                <p
                  class="text-2xl font-semibold  py-2"
                >
                  Hello 
                </p>
                <p
                  class="py-2"
                >
                  If you're coming here as an EthDenver Buidlathon winner, congratulations!
                </p>
                <p
                  class="py-2"
                >
                  Let's get you paid.
                </p>
                <p
                  class="py-2"
                >
                  Enter your email to stay up to date with next steps in claiming your award.
                </p>
                <p
                  class="py-2"
                >
                  OpenQ and EthDenver will use this email to tell you when you uploaded forms have been approved, after which you can claim right away.
                </p>
              </div>
              <section
                class="flex flex-col gap-3 pb-6"
              >
                <div
                  class="false py text-lg"
                >
                  <form
                    class="font-normal max-w-[500px] gap-4"
                  >
                    <div>
                      <div
                        class=" text-sm justify-between my-4"
                      >
                        <label
                          class="block font-semibold mb-1.5"
                          for="invoicingEmail"
                        >
                          Email 
                        </label>
                        <input
                          class="input-field w-full max-w-[400px] bg-input-bg px-3 py-[5px] border-border-muted invalid:border-danger"
                          id="invoicingEmail"
                          required=""
                          type="text"
                          value=""
                        />
                      </div>
                      <div
                        class="note my-4"
                      >
                        This email will be used by organizations to communicate with you. OpenQ will also send copies of any documents you send to organizers via our interface.
                      </div>
                    </div>
                    <input
                      class="btn-primary bg-green  text-sm py-1 px-2 text-center w-20 cursor-pointer"
                      type="submit"
                      value="Update"
                    />
                  </form>
                </div>
              </section>
              <p
                class="text-2xl font-semibold  py-2"
              >
                Now, tell us a bit about yourself
              </p>
              <div
                class="flex gap-2"
              >
                <p
                  class="font-semibold py-2"
                >
                  Username:
                </p>
                <div
                  class="flex items-center  "
                >
                  <div
                    class="flex-1 flex-row"
                  >
                    <div
                      class="flex w-fit items-center gap-4"
                    >
                      <div
                        class="font-normal flex-1"
                      />
                      <button
                        class="h-67"
                      >
                        <svg
                          class="fill-muted"
                          height="16"
                          viewBox="0 0 16 16"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M11.013 1.427a1.75 1.75 0 012.474 0l1.086 1.086a1.75 1.75 0 010 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 01-.927-.928l.929-3.25a1.75 1.75 0 01.445-.758l8.61-8.61zm1.414 1.06a.25.25 0 00-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 000-.354l-1.086-1.086zM11.189 6.25L9.75 4.81l-6.286 6.287a.25.25 0 00-.064.108l-.558 1.953 1.953-.558a.249.249 0 00.108-.064l6.286-6.286z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <p
                class="font-semibold py-2"
              >
                What is your role?
              </p>
              <button
                class="btn-default m-1 false"
              >
                Engineer
              </button>
              <button
                class="btn-default m-1 false"
              >
                Founder
              </button>
              <button
                class="btn-default m-1 false"
              >
                Product Manager
              </button>
              <button
                class="btn-default m-1 false"
              >
                Project Manager
              </button>
              <button
                class="btn-default m-1 false"
              >
                QA
              </button>
              <button
                class="btn-default m-1 false"
              >
                Designer
              </button>
              <button
                class="btn-default m-1 false"
              >
                Student
              </button>
              <button
                class="btn-default m-1 false"
              >
                Marketing
              </button>
              <button
                class="btn-default m-1 false"
              >
                Other
              </button>
              <p
                class="font-semibold pt-4 py-2"
              >
                Tell us about your skills
              </p>
              <div
                class="flex flex-wrap"
              />
              <div
                class="flex justify-center gap-2 mt-2"
              >
                <input
                  class="flex input-field w-full"
                  placeholder="Programming languages, frameworks..."
                  value=""
                />
                <button
                  class="btn-primary"
                >
                  Add
                </button>
              </div>
              <p
                class="font-semibold pt-4 py-2"
              >
                Which OpenQ products are you interested in?
              </p>
              <button
                class="btn-default m-1 false"
              >
                Marketplace
              </button>
              <button
                class="btn-default m-1 false"
              >
                Hackathon
              </button>
              <button
                class="btn-default m-1 false"
              >
                DRM
              </button>
              <button
                class="btn-default m-1 false"
              >
                Not sure
              </button>
              <div
                class="flex flex-col pt-8 py-2"
              >
                <div
                  class="flex items-center justify-start gap-2"
                >
                  <input
                    class="checkbox"
                    type="checkbox"
                  />
                  <p
                    class="font-semibold"
                  >
                    Join hundreds of professionals learning with OpenQ
                  </p>
                </div>
                <div
                  class="flex flex-col pt-4 text-muted gap-2"
                >
                  <p>
                    Curated emails on the latest software and web3 topics, product updates, events and new opportunities for your career.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex justify-between items-center py-2 px-4 flex-wrap border-t border-gray-700 w-full "
          >
            <div
              class="my-1"
            />
            <div>
              <button
                class="btn-primary false"
              >
                Continue
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="bg-overlay z-10 fixed inset-0"
    />
  </div>
</DocumentFragment>
`;
